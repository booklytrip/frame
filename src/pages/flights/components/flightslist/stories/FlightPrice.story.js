import React from 'react';
import { IntlProvider } from 'react-intl';
import { storiesOf } from '@storybook/react';
import centered from '@kadira/react-storybook-decorator-centered';

import FlightPrice from '../FlightPrice.jsx';

storiesOf('flights/FlightPrice', module)
    .addDecorator(story =>
        <IntlProvider locale="en">
            {story()}
        </IntlProvider>,
    )
    .addDecorator(centered)
    .add('FlightPrice', () =>
        <FlightPrice price={20} totalPrice={100} currency="EUR" />,
    );
