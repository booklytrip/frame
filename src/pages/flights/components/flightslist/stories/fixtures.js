import { merge } from 'lodash';

export const passengerPrice = {
    adult: {
        total: 110,
        tax: 10,
        base: 100,
    },
    child: {
        total: 55,
        tax: 5,
        base: 50,
    },
    infant: {
        total: 12,
        tax: 2,
        base: 10,
    },
    avg: 29.5,
};

export const passengerNumber = {
    adults: 1,
    children: 2,
    infants: 3,
};

export const oneWayFlight = {
    general: {
        currency: 'EUR',
        supplier: 'multiple_lc',
        price: {
            currency: 'EUR',
            amount: 191.67,
        },
        nightTransfer: true,
        pricing: passengerPrice,
        passengers: passengerNumber,
    },
    forwardSector: {
        departureAirport: {
            code: 'RIX',
            name: 'R朝ga',
            city: {
                name: 'Riga',
            },
        },
        arrivalAirport: {
            code: 'LGW',
            name: 'Gatwick',
            city: {
                name: 'Gatwick',
            },
        },
        departureTime: '2016-08-23 13:15:00',
        arrivalTime: '2016-08-24 15:35:00',
        stops: 2,
        duration: '84000',
        segments: [
            {
                departureAirport: {
                    code: 'RIX',
                    name: 'R朝ga',
                    city: {
                        name: 'Riga',
                    },
                },
                arrivalAirport: {
                    code: 'SXF',
                    name: 'Schoenefeld',
                    city: {
                        name: 'Berlin',
                    },
                },
                departureTime: '2016-08-23 13:15:00',
                arrivalTime: '2016-08-23 13:55:00',
                duration: '6000',
                stopDuration: null,
                flightNumber: 'FR2600',
                leftPlaces: 10,
                carrier: {
                    code: 'FR',
                    name: 'Ryanair',
                },
                supplier: 'ryanair',
            },
            {
                departureAirport: {
                    code: 'SXF',
                    name: 'Schoenefeld',
                    city: {
                        name: 'Berlin',
                    },
                },
                arrivalAirport: {
                    code: 'HHN',
                    name: 'Frankfurt',
                    city: {
                        name: 'Frankfurt',
                    },
                },
                departureTime: '2016-08-24 09:45:00',
                arrivalTime: '2016-08-24 10:35:00',
                duration: '6600',
                stopDuration: 3600,
                flightNumber: 'D82751',
                leftPlaces: null,
                carrier: {
                    code: 'DY',
                    name: 'Norwegian Air Shuttle',
                },
                supplier: 'norwegianair',
            },
            {
                departureAirport: {
                    code: 'HHN',
                    name: 'Frankfurt',
                    city: {
                        name: 'Frankfurt',
                    },
                },
                arrivalAirport: {
                    code: 'LGW',
                    name: 'Gatwick',
                    city: {
                        name: 'Gatwick',
                    },
                },
                departureTime: '2016-08-24 19:45:00',
                arrivalTime: '2016-08-25 00:35:00',
                duration: '6600',
                stopDuration: 7200,
                flightNumber: 'D82751',
                leftPlaces: null,
                carrier: {
                    code: 'DY',
                    name: 'Norwegian Air Shuttle',
                },
                supplier: 'norwegianair',
            },
        ],
    },
};

export const roundTripFlight = Object.assign({}, oneWayFlight, {
    comebackSegments: [
        {
            departureAirport: {
                code: 'LTN',
                name: 'Luton Airport',
                city: {
                    name: 'Luton',
                },
            },
            arrivalAirport: {
                code: 'RIX',
                name: 'R朝ga',
                city: {
                    name: 'Riga',
                },
            },
            departureTime: '2016-08-28 19:45:00',
            arrivalTime: '2016-08-29 00:20:00',
            duration: '9300',
            stopDuration: null,
            flightNumber: 'W62504',
            leftPlaces: null,
            carrier: {
                code: 'W6',
                name: 'Wizz Air',
            },
            supplier: 'wizzair',
        },
    ],
    comebackSector: {
        departureAirport: {
            code: 'LTN',
            name: 'Luton Airport',
            city: {
                name: 'Luton',
            },
        },
        arrivalAirport: {
            code: 'RIX',
            name: 'R朝ga',
            city: {
                name: 'Riga',
            },
        },
        departureTime: '2016-08-28 19:45:00',
        arrivalTime: '2016-08-29 00:20:00',
        stops: 0,
        duration: '9300',
    },
});

export const fewerSeatsFlight = merge({}, roundTripFlight, {
    forwardSector: {
        segments: [
            {
                leftPlaces: 2,
            },
        ],
    },
});
