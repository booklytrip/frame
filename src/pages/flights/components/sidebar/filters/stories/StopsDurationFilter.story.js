import React from 'react';
import { IntlProvider } from 'react-intl';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import centered from '@kadira/react-storybook-decorator-centered';

import StopsDurationFilter from '../StopsDurationFilter.jsx';

storiesOf('flights/sidebar/filters/StopsDurationFilter', module)
    .addDecorator(story =>
        <IntlProvider locale="en">
            {story()}
        </IntlProvider>,
    )
    .addDecorator(story =>
        <div style={{ width: '250px' }}>
            {story()}
        </div>,
    )
    .addDecorator(centered)
    .add('default', () =>
        <StopsDurationFilter
            minDuration={0}
            maxDuration={86400}
            step={300}
            onChange={action('onChange')}
        />,
    );
