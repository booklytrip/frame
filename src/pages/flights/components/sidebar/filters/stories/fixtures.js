/**
 * @flow
 */

import type {
    Airport,
    FlightStopsFilter,
    FlightCarrierFilter,
} from '../../../../../../types';

export {
    oneWayFlight,
    roundTripFlight,
} from '../../../flightslist/stories/fixtures';

export const stops: Array<FlightStopsFilter> = [
    {
        price: 600,
        stops: 1,
        enabled: true,
    },
    {
        price: 1200,
        stops: 2,
        enabled: false,
    },
    {
        price: 1800,
        stops: 3,
        enabled: true,
    },
];

export const carriers: Array<FlightCarrierFilter> = [
    {
        price: 600,
        carrier: {
            code: 'FR',
            name: 'Ryanair',
        },
        enabled: true,
    },
    {
        price: 1200,
        carrier: {
            code: 'W6',
            name: 'Wizz Air',
        },
        enabled: false,
    },
];

export const departureAirport: Airport =  {
    code: 'SXF',
    name: 'Schoenefeld',
    city: {
        code: 'BER',
        name: 'Berlin',
    },
};

export const arrivalAirport: Airport = {
    code: 'HHN',
    name: 'Frankfurt',
    city: {
        code: 'FRA',
        name: 'Frankfurt',
    },
};

export const flightDirection = {
    loading: false,
    departureAirport,
    arrivalAirport,
    minDuration: 2400,
    maxDuration: 20400,
};

export const searchQuery = {
    departureAirport: 'RIX',
    arrivalAirport: 'LON',
    departureDate: '2017-04-01',
    returnDate: '2017-04-01',
    adults: 1,
    children: 0,
    infants: 0,
};
