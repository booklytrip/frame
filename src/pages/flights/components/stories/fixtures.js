export const departureAirport = {
    code: 'RIX',
    name: 'Riga International Airport',
    city: {
        name: 'Riga',
    },
};

export const arrivalAirport = {
    code: 'STN',
    name: 'London Stansted Airport',
    city: {
        name: 'London',
    },
};
