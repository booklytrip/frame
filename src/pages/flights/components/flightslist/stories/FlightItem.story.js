import React from 'react';
import { IntlProvider } from 'react-intl';
import { storiesOf } from '@storybook/react';
import centered from '@kadira/react-storybook-decorator-centered';

import { oneWayFlight, roundTripFlight, fewerSeatsFlight } from './fixtures';

import { Container } from 'semantic-ui-react';
import FlightItem from '../FlightItem.jsx';

storiesOf('flights/FlightItem')
    .addDecorator(story =>
        <IntlProvider locale="en">
            {story()}
        </IntlProvider>,
    )
    .addDecorator(story =>
        <Container>
            {story()}
        </Container>,
    )
    .addDecorator(centered)
    .add('one-way flight', () => <FlightItem flight={oneWayFlight} />)
    .add('round-trip flight', () => <FlightItem flight={roundTripFlight} />)
    .add('fewer seats left', () => <FlightItem flight={fewerSeatsFlight} />);
