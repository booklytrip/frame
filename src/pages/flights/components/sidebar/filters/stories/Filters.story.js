import React from 'react';
import { IntlProvider } from 'react-intl';
import { storiesOf } from '@storybook/react';
import centered from '@kadira/react-storybook-decorator-centered';

import { ApolloProvider } from '../../../../../../lib/jest/apollo';

import Filters from '../Filters.jsx';

import { oneWayFlight, roundTripFlight, searchQuery } from './fixtures';

storiesOf('flights/sidebar/Filters', module)
    .addDecorator(story =>
        <ApolloProvider>
            {story()}
        </ApolloProvider>,
    )
    .addDecorator(story =>
        <IntlProvider locale="en">
            {story()}
        </IntlProvider>,
    )
    .addDecorator(story =>
        <div style={{ width: '250px' }}>
            {story()}
        </div>,
    )
    .addDecorator(centered)
    .add('loading', () => <Filters loading />)
    .add('with flights', () =>
        <Filters
            loading={false}
            flights={[oneWayFlight, roundTripFlight]}
            rawFlights={[oneWayFlight, roundTripFlight]}
            query={searchQuery}
        />,
    );
