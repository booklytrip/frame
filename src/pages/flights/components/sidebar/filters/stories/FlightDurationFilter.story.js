import React from 'react';
import { IntlProvider } from 'react-intl';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import centered from '@kadira/react-storybook-decorator-centered';

import FlightDurationFilter from '../FlightDurationFilter.jsx';

import { flightDirection } from './fixtures';

storiesOf('flights/sidebar/filters/FlightDurationFilter', module)
    .addDecorator(story =>
        <IntlProvider locale="en">
            {story()}
        </IntlProvider>,
    )
    .addDecorator(story =>
        <div style={{ width: '250px' }}>
            {story()}
        </div>,
    )
    .addDecorator(centered)
    .add('default', () =>
        <FlightDurationFilter
            directions={[flightDirection]}
            onChange={action('onChange')}
        />,
    )
    .add('loading', () =>
        <FlightDurationFilter
            directions={[
                {
                    ...flightDirection,
                    loading: true,
                },
            ]}
            onChange={action('onChange')}
        />,
    );
