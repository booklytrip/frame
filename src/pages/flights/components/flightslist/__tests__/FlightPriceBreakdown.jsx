import React from 'react';
import { mountWithIntl, createWithIntl } from '../../../../../lib/jest/intl';
import FlightPriceBreakdown from '../FlightPriceBreakdown';

import { passengerPrice, passengerNumber } from '../stories/fixtures';

describe('FlightPriceBreakdown', () => {
    it('renders without crashing', () => {
        mountWithIntl(
            <FlightPriceBreakdown
                totalPrice={{
                    amount: 100,
                    currency: 'EUR',
                }}
                pricePerPassenger={{
                    amount: 20,
                    currency: 'EUR',
                }}
                passengerPrice={passengerPrice}
                passengerNumber={passengerNumber}
            />,
        );
    });

    describe('Snapshots', () => {
        it('renders price breakdown', () => {
            const rendered = createWithIntl(
                <FlightPriceBreakdown
                    totalPrice={{
                        amount: 100,
                        currency: 'EUR',
                    }}
                    pricePerPassenger={{
                        amount: 20,
                        currency: 'EUR',
                    }}
                    passengerPrice={passengerPrice}
                    passengerNumber={passengerNumber}
                />,
            );

            expect(rendered.toJSON()).toMatchSnapshot();
        });
    });
});
