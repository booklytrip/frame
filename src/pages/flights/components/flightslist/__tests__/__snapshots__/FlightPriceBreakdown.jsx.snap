// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlightPriceBreakdown Snapshots renders price breakdown 1`] = `
<div
  className="ui basic segment hkiwyU"
>
  <div
    className="ui one column grid"
  >
    <div
      className="column fplSkJ"
    >
      <table
        className="ui small single line very basic compact table"
      >
        <thead
          className=""
        >
          <tr
            className=""
          >
            <th
              className=""
            >
              <span>
                Passengers
              </span>
            </th>
            <th
              className="right aligned"
            >
              <span>
                Price/passenger
              </span>
            </th>
            <th
              className="center aligned"
            >
              <span>
                Count
              </span>
            </th>
            <th
              className="right aligned"
            >
              <span>
                Price
              </span>
            </th>
          </tr>
        </thead>
        <tbody
          className=""
        >
          <tr
            className=""
          >
            <td
              className=""
            >
              Adults
            </td>
            <td
              className="right aligned"
            >
              € 
              110
            </td>
            <td
              className="center aligned"
            >
              1
            </td>
            <td
              className="right aligned"
            >
              € 
              110
            </td>
          </tr>
          <tr
            className=""
          >
            <td
              className=""
            >
              Children
            </td>
            <td
              className="right aligned"
            >
              € 
              55
            </td>
            <td
              className="center aligned"
            >
              2
            </td>
            <td
              className="right aligned"
            >
              € 
              110
            </td>
          </tr>
          <tr
            className=""
          >
            <td
              className=""
            >
              Infants
            </td>
            <td
              className="right aligned"
            >
              € 
              12
            </td>
            <td
              className="center aligned"
            >
              3
            </td>
            <td
              className="right aligned"
            >
              € 
              36
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      className="ui fitted divider kXzbfq"
      style={Object {}}
    />
    <div
      className="column eIgfYy"
    >
      <table
        className="ui small single line very basic compact table"
      >
        <tbody
          className=""
        >
          <tr
            className=""
          >
            <td
              className=""
              colSpan="3"
            >
              <span
                className="bBareZ"
              >
                <span>
                  Total
                </span>
              </span>
            </td>
            <td
              className="right aligned"
            >
              € 
              100
            </td>
          </tr>
          <tr
            className=""
          >
            <td
              className=""
              colSpan="3"
            >
              <span
                className="bBareZ"
              >
                <span>
                  Price per passenger
                </span>
              </span>
            </td>
            <td
              className="right aligned"
            >
              € 
              20
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
